### PR TITLE
Removing code that converted emails to user ids

### PR DESCRIPTION
**Pull request by decryan1984**
# Pull Request Template

## Description

The ediscovery service now manages all interactions with CI, and all code to convert emails to user ids and vice versa is now located there. Removing redundant code from plugin and making changes to handle the new interactions with ediscovery. All changes in the service have been made backwards compatible so that the existing version of the sdk (with email to id conversions) can operate as normal.

Fixes # (issue)
SPARK-54197

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Does not affect any existing tests and reduces the complexity of the plugin.

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js [hunk 1]
```diff
@@ -43,7 +42,24 @@ const EDiscovery = WebexPlugin.extend({
...
+              return Promise.reject(new InvalidEmailAddressError(invalidEmails));
+            }
+          }
+          catch (error) {
+            // assume syntax error and continue
+          }
+        }
+
+        return Promise.reject(reason);
+      });
```
Comment (decryan1984) [inline packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js]: We currently supply a pre-existing custom error to the Atlas UI if their request contained invalid emails, this is still the case and requires no change on their part

File: packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js [hunk 2]
```diff
@@ -220,109 +236,6 @@ const EDiscovery = WebexPlugin.extend({
...
-            if (emails.length !== validUserNames.length) {
-              // if user ids could not be found for any emails we assume they were invalid and throw a custom error
-              const invalidEmails = emails.filter((email) => !validUserNames.includes(email));
-
-              return Promise.reject(new InvalidEmailAddressError(invalidEmails));
-            }
-
-            return userIds;
-          });
-      });
```
Comment (decryan1984) [inline packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js]: This is all managed in the service now

File: packages/node_modules/@webex/internal-plugin-ediscovery/src/index.js [hunk 3]
```diff
@@ -221,6 +192,8 @@ registerInternalPlugin('ediscovery', EDiscovery, {
                     }
 
                     if (reportRequest.emails) {
+                      // store unencrypted emails for ediscovery service to convert to user ids
+                      reportRequest.unencryptedEmails = reportRequest.emails;
```
Comment (decryan1984) [inline packages/node_modules/@webex/internal-plugin-ediscovery/src/index.js]: The service will take these emails, convert to user ids, and remove them, ensuring they are never stored in the db

**Comment by taymoork2** [comment]
Do tests need to be updated for this or since the service is handling most of the logic now, tests can stay the same?

**Comment by decryan1984** [comment]
I haven’t added tests for this change, since we’ve really only removed logic to the service.

File: packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js [hunk 4]
```diff
@@ -43,7 +42,24 @@ const EDiscovery = WebexPlugin.extend({
...
+    })
+      .catch((reason) => {
+        // This value correspondes to the ediscovery error code INVALID_USER_EMAILS_LIST_IN_MESSAGE
+        if (reason.body.errorCode === 14400001) {
+          try {
+            const invalidEmails = JSON.parse(reason.body.message);
+
+            if (Array.isArray(invalidEmails) && invalidEmails.length) {
+              return Promise.reject(new InvalidEmailAddressError(invalidEmails));
+            }
```
Comment (gannons) [inline packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js]: Should you handle the scenario where it's not an array or the array is empty?

Likely you'll want to log a warn/error.

File: packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js [hunk 5]
```diff
@@ -43,7 +42,24 @@ const EDiscovery = WebexPlugin.extend({
...
+          }
+          catch (error) {
+            // assume syntax error and continue
```
Comment (gannons) [inline packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js]: Good idea to log an error

i.e. As this should never be hit

File: packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js [hunk 6]
```diff
@@ -43,7 +42,24 @@ const EDiscovery = WebexPlugin.extend({
       service: 'ediscovery',
       resource: 'reports',
       body
-    });
+    })
+      .catch((reason) => {
+        // This value correspondes to the ediscovery error code INVALID_USER_EMAILS_LIST_IN_MESSAGE
+        if (reason.body.errorCode === 14400001) {
```
Comment (mattnorris) [inline packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js]: This code may be better in a `constants` file with a friendlier name.

**Pull request merged**